### PR TITLE
feat: refresh button clears results cache

### PR DESCRIPTION
**Pull request by magnew**
Closes: #7718 

### Description:

This is a proof of concept for clearing the results cache with the refresh button. This doesn't have:

- a good way to turn caching back on

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/controllers/savedChartController.ts [hunk 1]
```diff
@@ -52,6 +52,8 @@ export class SavedChartController extends Controller {
                 req.user!,
                 chartUuid,
                 body.filters,
+                undefined,
+                req.query?.invalidateCache?.toString() === 'true',
```
Comment (magnew) [inline packages/backend/src/controllers/savedChartController.ts:56]: Is there a better way than this to get query parameters from the url?

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 2]
```diff
@@ -900,6 +901,7 @@ export class ProjectService {
                 : QueryExecutionContext.CHART,
 
             queryTags,
+            invalidateCache,
```
Comment (magnew) [inline packages/backend/src/services/ProjectService/ProjectService.ts:904]: These should all probably become named parameters

**Comment by owlas** [comment]
Could we make this a parameter of when we `mutate()` to get new results? i.e. on `runQuery()` we would do `runQuery(useCache?)`

File: packages/frontend/src/hooks/useQueryResults.tsx [hunk 3]
```diff
@@ -163,8 +167,17 @@ export const useUnderlyingDataResults = (
...
+export const useChartResults = (
+    chartUuid: string,
+    filters?: Filters,
+    invalidateCache?: boolean,
+) => {
+    const queryKey = [
+        'savedChartResults',
+        chartUuid,
+        JSON.stringify(filters),
+        invalidateCache,
```
Comment (owlas) [inline packages/frontend/src/hooks/useQueryResults.tsx:179]: @magnew do we actually need to track the cache in the state? I don't think we need the charts to rerun when this status changes. We just need the charts to re-run when the user hits refresh

Comment (magnew) [reply packages/frontend/src/hooks/useQueryResults.tsx:179]: Right, this is the part where we are fighting with react query a bit. When you hit refresh, we need to invalidate the queries. One way to do that is to put some cache indicator in the query key. That's why I'm keeping it here, but I would like to isolate it from the view code.

File: packages/backend/src/controllers/savedChartController.ts [hunk 4]
```diff
@@ -48,11 +48,14 @@ export class SavedChartController extends Controller {
...
-                req.user!,
+            results: await projectService.runViewChartQuery({
+                user: req.user!,
                 chartUuid,
-                body.filters,
-            ),
+                filters: body.filters,
+                versionUuid: undefined,
+                invalidateCache:
+                    req.query?.invalidateCache?.toString() === 'true',
```
Comment (magnew) [inline packages/backend/src/controllers/savedChartController.ts:57]: @ZeRego is there a better way to get this query arg? A decorator or something?

Comment (magnew) [reply packages/backend/src/controllers/savedChartController.ts]: Done

**Review by ZeRego — APPROVED** [review]

**Pull request merged**